File: cpp/2341.maximum-number-of-pairs-in-array.cpp
#include <vector>
using std::vector;

class Solution {
public:
  vector<int> numberOfPairs(vector<int> &nums) {
    const int n = 101;

    vector<int> pairs(n, 0);

    for (int num : nums) {
      pairs[num]++;
    }

    vector<int> res = {0, 0};
    for (int pair : pairs) {
      res[0] += pair / 2;
      res[1] += pair % 2;
    }

    return res;
  }
};

File: golang/2341.maximum-number-of-pairs-in-array.go
func numberOfPairs(nums []int) []int {
	const N = 101
	pairs := make([]int, N)

	for _, num := range nums {
		pairs[num]++
	}

	res := make([]int, 2)

	for _, p := range pairs {
		res[0] += p / 2
		res[1] += p % 2
	}

	return res
}
